src/Sarif/Visitors/GitHubIngestionVisitor.cs

```diff
@@ -12,42 +13,19 @@ public class GitHubIngestionVisitor : SarifRewritingVisitor
         // Internal static rather than private const to allow a unit test with a practical limit.
         internal static int s_MaxResults = 5000;
 
+        private Run run;
+        private int ruleIndex = -1;
         private IList<Artifact> artifacts;
         private IList<ThreadFlowLocation> threadFlowLocations;
 
         public override Run VisitRun(Run node)
         {
+            this.run = node;
             this.artifacts = node.Artifacts;
             this.threadFlowLocations = node.ThreadFlowLocations;
 
             if (node.Results != null)
             {
-                int errorsCount = 0;
-                foreach (Result result in node.Results)
-                {
-                    if (result.Level == FailureLevel.Error)
-                    {
-                        errorsCount++;
-                    }
-                }
-
-                if (errorsCount != node.Results.Count)
-                {
-                    var errors = new List<Result>();
-
-                    foreach (Result result in node.Results)
-                    {
-                        if (result.Level == FailureLevel.Error)
-                        {
-                            errors.Add(result);
-                        }
-
-                        if (errors.Count == s_MaxResults) { break; }
-                    }
-
-                    node.Results = errors;
-                }
```

Comment:
please also see the test file "NonErrorResults.sarif'.
This does not seem right, why do we want to remove warning results?

Reply:
Perhaps GitHub only used to process errors? Have you tested this in their system?

Reply:
This is from the current SDK repo:
![image](https://user-images.githubusercontent.com/81775155/206024875-c9a5d442-f6bc-4237-897d-f5beeb83d286.png)